test/encryption/encryption.test.js

```diff
@@ -1307,4 +1157,302 @@ describe('encryption integration tests', () => {
+      it('returns `null`', async function() {
+        const connection = mongoose.createConnection();
+        const model = connection.model('Name', { age: String });
+
+        await connection.openUri(process.env.MONGOOSE_TEST_URI);
+
+        assert.equal(model.clientEncryption(), null);
+      });
+    });
+
+    describe('Client not connected', function() {
+      it('returns `null`', async function() {
+        const connection = mongoose.createConnection();
+        const model = connection.model('Name', { age: String });
+
+        assert.equal(model.clientEncryption(), null);
+      });
+    });
+
+    describe('auto encryption enabled for a collection', function() {
+      let connection;
+      let model;
+      beforeEach(async function() {
+        connection = createConnection();
+        model = connection.model('Model1', new Schema({
+          a: {
+            type: String,
+            encrypt: {
+              keyId
+            }
+          }
+        }, {
+          encryptionType: 'queryableEncryption'
+        }));
+
+        await connection.openUri(process.env.MONGOOSE_TEST_URI, autoEncryptionOptions());
+
+        afterEach(async function() {
+          await connection.close();
+        });
+
+        it('returns a client encryption object', async function() {
+          assert.ok(model.clientEncryption() instanceof mdb.ClientEncryption);
+        });
+
+        it('the client encryption is usable as a key vault', async function() {
+          const clientEncryption = model.clientEncryption();
+          const dataKey = await clientEncryption.createDataKey('local');
+          const keys = await clientEncryption.getKeys().toArray();
+
+          assert.ok(keys.length > 0);
+
+          const key = keys.find(
+            ({ _id }) => _id.toString() === dataKey.toString()
+          );
+
+          assert.ok(key);
+        });
+
+        it('uses the same keyvaultNamespace', async function() {
+          assert.equal(model.clientEncryption()._keyVaultNamespace, 'keyvault.datakeys');
+        });
+
+        it('uses the same kms providers', async function() {
+          assert.deepEqual(model.clientEncryption()._kmsProviders, { local: { key: LOCAL_KEY } });
+        });
+
+        it('uses the same proxy options', async function() {
+          const options = model.collection.conn.client.options.autoEncryption;
+          options.proxyOptions = { name: 'bailey' };
+          assert.deepEqual(model.clientEncryption()._proxyOptions, { name: 'bailey' });
+        });
+
+        it('uses the same TLS options', async function() {
+          const options = model.collection.conn.client.options.autoEncryption;
+          options.tlsOptions = {
+            tlsCAFile: 'some file'
+          };
+          assert.deepEqual(model.clientEncryption()._tlsOptions, {
+            tlsCAFile: 'some file'
+          });
+        });
+
+        it.skip('uses the same credentialProviders', async function() {
```

Comment:
This is a feature of 6.15+ drivers - is this something you plan to upgrade to soon?  If not, I can remove this test. If we do adopt 6.15, then I'll leave this test in and we can unskip it when the time comes.

Reply:
I will ship a  Mongoose release that uses Node driver 6.16 this week.